docs/DeveloperGuide.md

```diff
@@ -767,21 +767,41 @@ The following sequence diagrams shows how the unmark command works:
+1. When the user inputs "clear", the `LogicManager` parses the given input text using `AddressBookParser#parseCommand()`.
 2. A new `ClearCommand` is created by `AddressBookParser` and returned to the `LogicManager`.
-3. The `LogicManager` will then call `ClearCommand#execute(Model model)`, which returns a new `CommandResult` object (with its `isClearRequest` field set to `true`) to the `LogicManager`.
+3. The `LogicManager` calls `ClearCommand#execute(Model model)`, which returns a new `CommandResult` object (with its `isClearRequest` field set to `true`) to the `LogicManager`.
 4. The `CommandResult` (with its `isClearRequest` field set to `true`) is then returned to the `MainWindow`.
 5. To confirm the user's decision to clear TAPA, the `MainWindow` executes `MainWindow#handleClearRequest()` which updates the `CommandBox` to only recognise a `confirm` input for the next command.
-    <div markdown="span" class="alert alert-info">:information_source:
-    <b>Note:</b> A `CommandException` will be thrown if the user inputs anything other than confirm for the next input. The CommandBox then returns to accepting all commands as described in the user guide.
-    </div>
-6. If the user inputs `confirm`, `LogicManager` parses the given input text using `AddressBookParser#parseCommand()` and a new `ConfirmClearCommand` is created and returned to the `LogicManager`.
+
+![ClearCommandSequenceDiagram](images/ClearCommandSequenceDiagram.png)
+
+<div markdown="span" class="alert alert-info">:information_source:
+<b>Note:</b> A `CommandException` will be thrown if the user inputs anything other than `confirm` for the next input. The `CommandBox` will then return to its normal operation, accepting all commands as described in the user guide.
+</div>
+
+![ClearCommandActivityDiagram](images/ClearCommandActivityDiagram.png)
+
+6. If the user inputs "confirm", the `LogicManager` parses the given input text using `AddressBookParser#parseCommand()`, then a new `ConfirmClearCommand` is created and returned to the `LogicManager`.
 7. The `LogicManager` will then call `ConfirmClearCommand#execute(Model model)`.
-8. In the `ConfirmClearCommand`, `model.setAddressBook(new AddressBook())` is invoked, resetting the current list of students stored in TAPA to an empty list.
-9. Lastly, the `ConfirmClearCommand` will create a new `CommandResult`, which will be returned to the `LogicManager`.
+8. The `ConfirmClearCommand` calls `Model#setAddressBook(new AddressBook())` which resets the current list of students stored in TAPA to an empty list.
+9. Lastly, the `ConfirmClearCommand` creates a new `CommandResult`, which is returned to the `LogicManager`.
+
+![ConfirmClearCommandSequenceDiagram](images/ConfirmClearCommandSequenceDiagram.png)
+
+#### Design considerations
+
+**Aspect: How TAPA confirms the user's decision to clear TAPA**
+
+* **Alternative 1 (current choice):** Have the user input `confirm`, which is treated by TAPA as a command being executed.
```

Comment:
```suggestion
* **Alternative 1 (current choice):** Have the user input "confirm", which is treated by TAPA as a command being executed.
```

@brelkh Hi, sorry but I missed out this one!

Reply:
Alright, I'll add this in too! 👍